src/test/ui/generic-associated-types/issue-80626.rs

```diff
@@ -0,0 +1,60 @@
+#![feature(generic_associated_types)]
+
+trait Allocator {
+    type Allocated<T>;
+    //~^ HELP consider relaxing
+    //~| HELP consider relaxing
+}
+
+enum LinkedList<A: Allocator> {
+    Head,
+    Next(u8, A::Allocated<Self>), //~ ERROR overflow evaluating the requirement
+}
+
+enum LinkedList2<A: Allocator> {
+    Head,
+    Next(A::Allocated<Self>, u8), //~ ERROR overflow evaluating the requirement
+}
+
+impl Allocator for () {
+    type Allocated<T> = Box<T>;
+}
+
+fn main() {
```

Comment:
I don't think we need to test runtime behavior here; this can just be empty with a `// check-pass`